Microsoft.Dynamics365.UIAutomation.Browser/Extensions/SeleniumExtensions.cs

```diff
@@ -70,14 +71,27 @@ public static void Hover(this IWebElement element, IWebDriver driver, bool ignor
 
         #endregion Click
 
+        public static void Click(this IWebDriver driver, IWebElement element, Func<Point> offsetFunc = null, bool ignoreStaleElementException = true)
+            => driver.Perform(a => a.Click(), element, offsetFunc, ignoreStaleElementException);
+
         #region Double Click
 
-        public static void DoubleClick(this IWebDriver driver, IWebElement element, bool ignoreStaleElementException = false)
+        public static void DoubleClick(this IWebDriver driver, IWebElement element, Func<Point> offsetFunc = null, bool ignoreStaleElementException = true)
+            => driver.Perform(a => a.DoubleClick(), element, offsetFunc, ignoreStaleElementException);
+
+        public static void Perform(this IWebDriver driver, Func<Actions, Actions> action, IWebElement element, Func<Point> offsetFunc = null, bool ignoreStaleElementException = true)
         {
             try
             {
-                Actions actions = new Actions(driver);
-                actions.DoubleClick(element).Build().Perform();
+                var actions = new Actions(driver);
+                if (offsetFunc == null)
+                    actions = actions.MoveToElement(element);
+                else
+                {
+                    var offset = offsetFunc();
+                    actions = actions.MoveToElement(element, offset.X, offset.Y);
+                }
+                action(actions).Perform();
```

Comment:
The Logic for the Click or DoubleClick in a position related to an inner element.